internal/storage/file_storage.go

```diff
@@ -15,7 +16,7 @@ type FileStorage struct {
 	mx       sync.RWMutex
 }
 
-func NewFileStorage(fileName string, store *MapStorage) *FileStorage {
+func NewFileStorage(fileName string, store *MapStorage) Storage {
```

Comment:
Нет смысла возвращать здесь ссылку на интерфейс, стоит вернуть указатель на тип реализации.
```suggestion
func NewFileStorage(fileName string, store *MapStorage) *FileStorage {
```